text/0000-inferred-types.md

```diff
@@ -81,98 +89,98 @@ impl Into<MyEnum> for Example {
     }
 }
 
-fn my_function(data: MyEnum) { /* ... */ }
+fn use_numbers(number: Numbers) { /* ... */ }
 
+use_numbers(Example::Alpha.into());
+//          ^^^^^^^^^^^^^^^^^^^^^ Valid code
 
-my_function(Example::Alpha.into()); // ✅
-
-
-my_function(_::Alpha().into()); // ❌ 
+use_numbers(_::Alpha.into());
+//          ^^^^^^^^^^^^^^^^^ error[E0599]: no variant or associated item named `Alpha` found for enum `Numbers` in the current scope
 ```
 
-
 # Reference-level explanation
-[reference-level-explanation]: #reference-level-explanation
 
-The `_` token can be used to simplify writing the type name explicitly when the type of its containing expression is known. The elided type does not need to be imported into scope. The `_` token does not allow access to implementations and trait implementations of any sort.
+[reference-level-explanation]: #reference-level-explanation
 
-Mentioned above, implementations of any sort are not allowed
-```rust
-#[derive(Default)]
+The underscore (`_`) syntax is designed to allow concise writing of rust code. Most of the time, it will just like when the type is written in a verbose way. The compiler will first look for the underscore token (`_`) followed by, optionally, type parameters (`::</* type & lifetime parameters */>`), then brackets (for `struct`s) or a variant (for `enum`s). Below are some variations of valid syntax.  
+  
+```rust  
+_::<&'static str>::EnumVariant(&"Hello, rust")  
```

Comment:
I think what we need to explain here is not how to parse the `_`, but in which positions (e.g. struct literal identifier, path parts, etc) would an underscore be valid.

Reply:
@SOF3, thanks again for all the feedback! I do need a bit of help here, I don't understand.

Reply:
@SOF3, can you explain what you mean?